lib/puppet/functions/simp_pki_service/validate_ca_hash.rb

```diff
@@ -0,0 +1,86 @@
+# Validate that the passed Hash of CAs meets the following requirements:
+#
```

Comment:
The explicit structure of the hash is not documented anywhere. So, I think it would be best to be very explicit here.   For example 'Root CA'  <==> CA hash for which 'root_ca' key has a true value